service/src/main/java/com/ericsson/eiffel/remrem/generate/controller/RemremGenerateController.java

```diff
@@ -54,17 +58,23 @@
      * <p>
      * <p>
      * Parameters:
-     * mp - The message protocol , which tells us which service to invoke.
+     * msgProtocol - The message protocol, which tells us which service to invoke.
      * msgType - The type of message that needs to be generated.
      * bodyJson - The content of the message which is used in creating the event details.
      * <p>
      * Returns:
      * The event information as a json element
      */
-    @RequestMapping(value = "/{mp}", method = RequestMethod.POST)
-    public ResponseEntity<?> generate(@PathVariable String mp, @RequestParam("msgType") String msgType,
-                                      @RequestBody JsonObject bodyJson) {
-        MsgService msgService = getMessageService(mp);
+    @ApiOperation(value = "To generate eiffel event based on the message protocol", response = String.class)
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "Event sent successfully"),
+            @ApiResponse(code = 400, message = "Malformed JSON"),
+            @ApiResponse(code = 500, message = "Internal server error"),
+            @ApiResponse(code = 503, message = "Message protocol is invalid") })
+    @RequestMapping(value = "/{mp" + REGEX + "}", method = RequestMethod.POST)
+    public ResponseEntity<?> generate(@ApiParam(value = "message protocol") @PathVariable("mp") String msgProtocol,
```

Comment:
better to have variable description, like message protocol instead of mp

Reply:
I did it in previous commits? but regarding to Vinod comment I can not change name of PathVariable from mp to message protocol. So I have added description to variables using ApiParam.. In swagger UI wecan see this description near field.